client.go

```diff
@@ -15,24 +15,52 @@ Provides a high-level client to talk to the API that football-data.org offers.
 
 To create an instance please use NewClient(h).
 */
-type Client struct {
-	h *http.Client
+type Client interface {
+	Fixture(id uint64) FixtureRequest
+	Fixtures() FixturesRequest
+	SoccerSeason(id uint64) SoccerSeasonRequest
+	FixturesOfSoccerSeason(soccerSeasonId uint64) SoccerSeasonFixturesRequest
+	LeagueTableOfSoccerSeason(soccerSeasonId uint64) SoccerSeasonLeagueTableRequest
+	TeamsOfSoccerSeason(soccerSeasonId uint64) SoccerSeasonTeamsRequest
+	SoccerSeasons() SoccerSeasonsRequest
+	Team(id uint64) TeamRequest
+	FixturesOfTeam(id uint64) TeamFixturesRequest
+
+	SetToken(authToken string)
+}
+
+/*
+Provides a high-level client implementation to talk to the API that football-data.org offers.
+
+To create an instance please use NewClient(h).
+*/
+type client struct {
```

Comment:
Not anymore instantiable from the outside. Hence, factory function is garanted to be the sole entry point.